gwpy/time/_tconvert.py

```diff
@@ -233,12 +233,16 @@ def _str_to_datetime(datestr):
     try:  # try known string
         return DATE_STRINGS[str(datestr).lower()]()
     except KeyError:  # any other string
-        try:
-            return dateparser.parse(datestr)
-        except (ValueError, TypeError) as exc:
-            exc.args = ("Cannot parse date string {0!r}: {1}".format(
-                datestr, exc.args[0]),)
-            raise
+        try:  # use maya
+            import maya
+            return maya.when(datestr).datetime()
```

Comment:
@duncanmmacleod, this is a potentially awesome general-purpose feature, maybe we should move it upstream to `from_gps`?

Reply:
@alurban, I think I may have misspoke when we talked on Slack. This code is as 'upstream' as it gets, i was referring to the printing of timezone-aware strings as the output of `python -m gwpy.time`, however, since `gwpy.time` (the module) just returns `datetime.datetime` objects users can manually transition to other timezones if they like.